### DISCUSSION TITLE
Can we use extensions from visualstudio market place for Code Server?

### DISCUSSION THREAD

**Discussion by sakthivel-axim** [coder/code-server#6077] [General] [answered]
As mentioned in FAQ (https://github.com/coder/code-server/blob/main/docs/FAQ.md#why-cant-code-server-use-microsofts-extension-marketplace), we cannot use any extension from https://marketplace.visualstudio.com . Is my understanding right?

If we search any extension in code server extensions sidebar, extension results are from which market place ?

Which is open source extension market place for Code Server?

**Comment by code-asher** [comment]
That is correct.  The marketplace we use is Open VSX (https://open-vsx.org).

**Comment by sakthivel-axim** [comment]
Thanks @code-asher . So, can build the extension from source and use it in the Code Server ?

**Reply by code-asher** [reply]
If the source is available and the license permits it, then yup you can.

**Reply by code-asher** [reply, marked as answer]
Note though that some extensions may have a runtime license that is different from the source code license.

For example, vscode-cpptools's license is MIT but they have runtime licenses that disallows using the binaries the extension downloads in anything except Microsoft products.

https://github.com/microsoft/vscode-cpptools/blob/main/LICENSE.md
https://github.com/microsoft/vscode-cpptools/blob/main/RuntimeLicenses/cpptools-LICENSE.txt

**Reply by sakthivel-axim** [reply]
Thanks @code-asher. Looks we cannot use vscode-cpptools extension source also. 

Any suggestion for C/C++ extension that works with code server and supports ARM (Beaglebone) ?

**Reply by code-asher** [reply]
The only open source C++ extension I know of is https://open-vsx.org/extension/llvm-vs-code-extensions/vscode-clangd but I have never used it.

**Reply by sakthivel-axim** [reply]
Thanks @code-asher
